src/rgw/rgw_main.cc

```diff
@@ -653,6 +653,23 @@ static int process_request(RGWRados *store, RGWREST *rest, RGWRequest *req, RGWC
 
   dout(1) << "====== req done req=" << hex << req << dec << " http_status=" << http_ret << " ======" << dendl;
 
+  switch(http_ret/100) {
+    case 1:
+      perfcounter->inc(l_rgw_http_status_1xx);
+      break;
+    case 2:
+      perfcounter->inc(l_rgw_http_status_2xx);
+      break;
+    case 3:
+      perfcounter->inc(l_rgw_http_status_3xx);
+      break;
+    case 4:
+      perfcounter->inc(l_rgw_http_status_4xx);
+      break;
+    case 5:
+      perfcounter->inc(l_rgw_http_status_5xx);
+      break;
+  }
```

Comment:
http_ret can also have the value zero in case of unknown error. Add another counter l_rgw_http_status_null to capture the default case. This will have the nice invariant that the sum of the counters will be equal to the number of requests since uptime.